backend/src/utilities/feature-extractor.js

```diff
@@ -93,25 +99,66 @@ async function processProductObjectAndInsertIntoDB(productObject) {
 
 function extractFeaturesListsFromProductObject(productObject) {
   return {
-    name: extractFeaturesFromValue(productObject.name),
-    category: extractFeaturesFromValue(productObject.category),
-    subcategory: extractFeaturesFromValue(productObject.subcategory)
+    name: extractFeaturesFromValue(productObject.name)
   };
 }
 
 function extractFeaturesFromValue(value) {
-  const lowerCaseValue = value.toLowerCase();
+  /* normalize any special characters/spaces that might effect feature matching.
+   offset map required to ensure "median index" is based off unnormalized string
+   */
+  const { normalizedValue, removeList } = removeSpecialCharactersAndGenerateOffsetList(value.toLowerCase());
+
+  const regex =
+    (normalizedValue.match(COMMA_DELIMITED_REGEX) || []).length >= COMMA_REGEX_THRESHOLD
+      ? COMMA_DELIMITED_REGEX
+      : STANDARD_FEATURE_REGEX;
+  return extractFeaturesByRegex(normalizedValue, regex, removeList).map(normalizeFeatureAfterExtraction);
+}
+
+function removeSpecialCharactersAndGenerateOffsetList(value) {
+  const removeList = [{ index: -1, offset: 0 }];
+  let offset = 0;
+  let matchResult = REMOVE_REGEX.exec(value);
+  while (matchResult !== null) {
+    const removeIndex = matchResult.index;
+
```

Comment:
I think one exception we should make here is for size ranges (i.e. 5-6). We should check if chars to both sides (excluding spaces) are integers and preserve the dash if that is the case.

Reply:
Done with the ranges